python/lsst/obs/base/mapping.py

```diff
@@ -145,14 +147,16 @@ def map(self, mapper, dataId, write=False):
         addFunc = "add_" + self.datasetType  # Name of method for additionalData
         if hasattr(mapper, addFunc):
             addFunc = getattr(mapper, addFunc)
-            additionalData = addFunc(actualId)
-            assert isinstance(additionalData, dict), "Bad type for returned data"
+            additionalData = addFunc(self.datasetType, actualId)
+            assert isinstance(additionalData, PropertySet), \
+                   "Bad type for returned data: %s" (type(additionalData),)
         else:
-            additionalData = actualId.copy()
+            additionalData = None
```

Comment:
I'm a bit worried by this.  I think some serializers/deserializers that RHL/Merlin have been writing may be depending on having access to the dataId this way.  Did you detect any such usage?

(I'm also a little worried that exposing the dataId in the ButlerLocation makes it too tempting to just do everything in a serializer, turning the entire Butler and Mapper into essentially a no-op, but I think that's already mostly or completely possible, so it's water under the bridge until the ButlerWG completes its work.)

Reply:
This change does not actually change the behaviour at all, because the contents of the `dataId` are copied into the `additionalData` [in `ButlerLocation.__init__`](https://github.com/lsst/daf_persistence/blob/master/python/lsst/daf/persistence/butlerLocation.py#L212-L213), so putting it in here was unnecessary.
There are tests in obs_base for this functionality and they still pass (I had to modify them though, because there's now more in the `additionalData`, but I did check that what used to be in them is still in them).